openqaoa/optimizers/optimization_methods.py

```diff
@@ -428,3 +428,132 @@ def stochastic_grad_descent(fun, x0, jac, args=(), stepsize=0.001, mass=0.9, sta
 
     return OptimizeResult(x=x, fun=fun(x), jac=g, nit=i+1, nfev=i+1, success=True)
 '''
+
+
+
+def CANS(fun, x0, args=(), maxfev=None, stepsize=0.00001, n_shots_min=10, n_shots_max=10000, mu=0.99, b=1e-06,
+                 maxiter=100, tol=10**(-6), jac=None, jac_w_variance=None,callback=None, **options): #jac_w_uncertainty=None
+
+    print('hi')
+
+    chi = np.zeros(len(x0))
+    xi = 0
+    n_shots = n_shots_min
+
+    bestx = x0
+    besty = np.real(fun(bestx, *args))
+    funcalls = 1  # tracks no. of function evals.
+    niter = 0
+    improved = True
+    stop = False
+
+    testx = np.copy(bestx)
+    testy = besty
+    while improved and not stop and niter < maxiter:
+        lipschitz = np.sum(np.abs(testx)) #TODO there is something wrong with this const
+
+        if not stepsize < 2/lipschitz:
```

Comment:
Should I save the number of shots at each step in the Logger? Otherwise the user can't know how many shots have been used